src/qibolab/_core/instruments/qm/components/configs.py

```diff
@@ -2,9 +2,19 @@
 
 from pydantic import Field
 
-from qibolab._core.components import AcquisitionConfig, DcConfig
+from qibolab._core.components import AcquisitionConfig, DcConfig, OscillatorConfig
 
-__all__ = ["OpxOutputConfig", "QmAcquisitionConfig", "QmConfigs"]
+__all__ = [
+    "OpxOutputConfig",
+    "QmAcquisitionConfig",
+    "QmConfigs",
+    "OctaveOscillatorConfig",
+    "OctaveOutputModes",
```

Comment:
Do you really need it to be accessible outside? (out of the `OctaveOscillatorConfig` scope)

Reply:
It is not supposed to be user facing but it is used internally in another file, in particular here:
https://github.com/qiboteam/qibolab/blob/c4f3e7a9b5babbd6efc881a9c0e4cac8fc1af05e/src/qibolab/_core/instruments/qm/config/devices.py#L8

Reply:
I don't have a clear rule to discriminate all the cases, but a limit case is whether the only module using something is in the exact same subpackage.

In that case, you would only import from the module itself, and never from the containing package. In that case, `__all__` is ignored (as in all direct import).

I would suggest to reserve `__all__` for features to be exposed to the user, such that we can always apply the submodules' `__all__` concatenation strategy in all `__init__.py`, without any dedicated selection. And instead using the leading `_` convention to distinguish what is private to a given module, and what should be available to other internal modules within the package, but not to the Qibolab's user.

True that, for deeply nested modules, used by other modules very far, it may be convenient to shorten a bit the import path, by lifting with re-exports the containing subpackage.
However, for a single level, we can import them manually, doing something like:
```py
# a.py
__all__ = [A, B]

A = 42
B = 43
C = 44
D = 45
_E = 46

# __init__.py
from .a import *
from .a import C
```
where `A` and `B` are supposed to be used by the Qibolab user, `D` is separately shortened, `D` is accessible, and `_E` is private.

But that's just a proposal.

Reply:
> In that case, you would only import from the module itself, and never from the containing package. In that case, `__all__` is ignored (as in all direct import).
> 
> I would suggest to reserve `__all__` for features to be exposed to the user, such that we can always apply the submodules' `__all__` concatenation strategy in all `__init__.py`, without any dedicated selection.

Sounds good to me, so I fixed it in ee6d41b41aea6eb7bfb88f31b598fda6faae5ff8.